libs/blocks/gnav/gnav-search.js

```diff
@@ -74,9 +74,9 @@ const getSuggestions = (json) => {
 };
 
 const onSearchInput = async ({ value, resultsEl, locale, searchInputEl }) => {
-  const results = await fetchResults(value, locale);
+  const results = await fetchResults(value, window.adobeid?.locale);
```

Comment:
I'd add a comment here to document why `locale` isn't a good choice and `window.adobeid?.locale` needs to be used instead.

Reply:
Some places use hyphen (en-US), and some use underscore (en_US). If we can't standardize it, we may keep both in the `locale` object.

Reply:
I'm still not sure I'd base this on another piece of logic running. `window.adobeid` is currently declared in the `decorateProfile` method [here](https://github.com/adobecom/milo/blob/ed5955c1ae1db4ab34f7bfad216893cdf58db230/libs/blocks/gnav/gnav.js#L464). But maybe the current page doesn't need a Profile, so that logic would never run, thus `adobeid.locale` would not get defined. We already have access to `locale` via the method's parameters, I think it would be more straightforward to use `locale?.ietf?.replace('-', '_')` here.

On another note, I made a few calls with both `en-US` and `en_US`, `de-CH` and `de_CH`. The results are identical, so I'm wondering why this change is actually needed.

Reply:
Dexter uses locales with an underscore.